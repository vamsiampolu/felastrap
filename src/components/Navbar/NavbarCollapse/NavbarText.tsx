import { ComponentPropsWithoutRef } from "react";
import { useFela } from "react-fela";

const base = () => ({
  paddingTop: "0.5rem",
  paddingBottom: "0.5rem",
  color: "rgba(0, 0, 0, 0.65)",
});

export type SpanProps = Omit<ComponentPropsWithoutRef<"span">, "children">;

export const NavbarText = ({
  className = "",
  text,
  ...props
}: { text: string } & SpanProps) => {
  const { css } = useFela();
  const cx = `${css(base)} navbar-text ${className}`.trim();
  return (
    <span data-testid="navbar-text" className={cx} {...props}>
      {text}
    </span>
  );
};
