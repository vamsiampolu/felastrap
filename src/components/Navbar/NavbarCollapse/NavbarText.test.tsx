import { IRenderer } from "fela";
import { NavbarText } from "./NavbarText";
import {
  Wrapper,
  createRenderer,
  createSnapshot,
  cssReboot,
} from "../../../test-helpers";
import { render, screen } from "@testing-library/react";

describe("components/Navbar/NavbarCollapse/NavbarText", () => {
  let renderer: IRenderer;

  beforeEach(() => {
    renderer = createRenderer();
    cssReboot(renderer);
  });

  afterEach(() => {
    renderer.clear();
  });

  it("should render a navbar text", async () => {
    const component = <NavbarText text="Navbar Text" />;
    render(<Wrapper renderer={renderer}>{component}</Wrapper>);

    expect(screen.getByTestId("navbar-text")).toBeInTheDocument();
    expect(screen.getByTestId("navbar-text")).toHaveClass("navbar-text");
    expect(screen.getByTestId("navbar-text")).toHaveTextContent("Navbar Text");

    const snapshot = await createSnapshot({ component, renderer });
    expect(snapshot).toMatchSnapshot();
  });
});
