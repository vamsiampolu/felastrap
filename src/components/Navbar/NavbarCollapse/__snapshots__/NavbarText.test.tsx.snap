// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Navbar/NavbarCollapse/NavbarText should render a navbar text 1`] = `
".a {
  padding-top: 0.5rem;
}
.b {
  padding-bottom: 0.5rem;
}
.c {
  color: rgba(0, 0, 0, 0.65);
}


<span data-testid="navbar-text" class="a b c navbar-text">Navbar Text</span>;
"
`;
